packages/svelte/scripts/generate-dts.js

```diff
@@ -1,14 +1,29 @@
+import * as fs from 'fs';
 import { createBundle } from 'dts-buddy';
 
+fs.readdirSync('src/runtime', { withFileTypes: true })
+	.filter((dirent) => dirent.isDirectory())
+	.forEach((dirent) => fs.writeFileSync(`${dirent.name}.d.ts`, `import 'types/index.d.ts';`));
+
+fs.writeFileSync('index.d.ts', `import 'types/index.d.ts';`);
+fs.writeFileSync('compiler.d.ts', `import 'types/index.d.ts';`);
+
+// TODO: some way to mark these as deprecated
+fs.mkdirSync('types/compiler', { recursive: true });
+fs.writeFileSync('types/compiler/preprocess.d.ts', `import '../index.d.ts';`);
+fs.writeFileSync('types/compiler/interfaces.d.ts', `import '../index.d.ts';`);
+
 await createBundle({
 	output: 'types/index.d.ts',
 	modules: {
 		svelte: 'src/runtime/public.d.ts',
 		'svelte/compiler': 'src/compiler/public.d.ts',
 		'svelte/types/compiler/preprocess': 'src/compiler/preprocess/public.d.ts',
+		'svelte/types/compiler/interfaces': 'src/compiler/interfaces.d.ts',
```

Comment:
What uses this?

Reply:
vite plugin svelte and at least one other thing on the CI list

Reply:
we should update them to use `svelte/compiler` as soon as v4 is out, so that these entry points can be dropped in v5